moha/hamiltonians.py

```diff
@@ -162,6 +162,21 @@ def generate_one_body_integral(self, basis: str, dense: bool):
         # check if elements in connectivity matrix are float
         elif np.all([isinstance(elem[2], float)
                     for elem in self.connectivity]):
+            if self.orbital_overlap is not None:
+                # Assuming self.orbital_overlap should be a 2D array
+                if hasattr(
+                        self,
+                        'orbital_overlap') and self.orbital_overlap.ndim == 2:
```

Comment:
I'm not quite sure what are these checks for:
1. we check if user provided orbital overlap -- that's correct
2. we check if class has attribute `orbital_overlap`. This part confuses me a bit. It looks like the previous check already assumes that `orbital_overlap` attribute is part the PPP class
3. We check if orbital_overlap is a 2d array. This check can be merged with the next line as: `if self.orbital_ovelap.shape != (self.n_sites, self.n_sites)`